libass/ass_font.c

```diff
@@ -26,15 +26,167 @@
 #include FT_GLYPH_H
 #include FT_TRUETYPE_TABLES_H
 #include FT_OUTLINE_H
+#include FT_TRUETYPE_IDS_H
 #include <limits.h>
 
+/**
+ * We don't exclude Cygwin for Windows since we use WideCharToMultiByte only,
+ * this shall not violate any Cygwin restrictions on Windows APIs.
+ */
+#if defined(_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#elif defined(CONFIG_ICONV)
+#include <iconv.h>
+#endif
+
 #include "ass.h"
 #include "ass_library.h"
 #include "ass_font.h"
 #include "ass_fontselect.h"
 #include "ass_utils.h"
 #include "ass_shaper.h"
 
+/**
+ *  Get the mbcs codepoint from the output bytes of iconv/WideCharToMultiByte,
+ *  by treating the bytes as a prefix-zero-byte-omitted big-endian integer.
+ */
+static uint32_t pack_mbcs_bytes(const char *bytes, size_t length)
```

Comment:
Thanks! I wonder whether this is worth marking `static inline` or it doesn’t make any difference.